apps/admin-portal/src/components/users/add-user-groups.tsx

```diff
@@ -0,0 +1,198 @@
+                    filteredGroupList.push(group);
+                    handleGroupListChange(filteredGroupList);
+                }
+            });
+        } else {
+            handleGroupListChange(initialValues?.initialGroupList);
+        }
+    };
+
+    const addGroups = () => {
+        const addedGroups = [ ...initialValues.tempGroupList ];
+        if (checkedUnassignedListItems?.length > 0) {
+            checkedUnassignedListItems.map((group) => {
+                if (!(initialValues.tempGroupList.includes(group))) {
+                    addedGroups.push(group);
+                }
+            });
+        }
+        handleTempListChange(addedGroups);
+        handleGroupListChange(initialValues.groupList.filter(x => !addedGroups.includes(x)));
+    };
+
+    const removeGroups = () => {
+        const removedGroups = [ ...initialValues.groupList ];
+        if (checkedAssignedListItems?.length > 0) {
+            checkedAssignedListItems.map((group) => {
+                if (!(initialValues.groupList.includes(group))) {
+                    removedGroups.push(group);
+                }
+            });
+        }
+        handleGroupListChange(removedGroups);
+        handleTempListChange(initialValues.tempGroupList.filter(x => !removedGroups.includes(x)));
+        setCheckedAssignedListItems(checkedAssignedListItems.filter(x => !removedGroups.includes(x)))
+    };
+
+    const handleUnassignedItemCheckboxChange = (group) => {
+        const checkedGroups = [ ...checkedUnassignedListItems ];
+
+        if (checkedGroups.includes(group)) {
+            checkedGroups.splice(checkedGroups.indexOf(group), 1);
+            setCheckedUnassignedListItems(checkedGroups);
+        } else {
+            checkedGroups.push(group);
+            setCheckedUnassignedListItems(checkedGroups);
+        }
+    };
+
+    const handleAssignedItemCheckboxChange = (group) => {
+        const checkedGroups = [ ...checkedAssignedListItems ];
+
+        if (checkedGroups.includes(group)) {
+            checkedGroups.splice(checkedGroups.indexOf(group), 1);
+            setCheckedAssignedListItems(checkedGroups);
+        } else {
+            checkedGroups.push(group);
+            setCheckedAssignedListItems(checkedGroups);
+        }
+    };
+
+    const createItemLabel = (roleName: string) => {
+
+        const role = roleName.split("/");
+        if (role.length > 0) {
+            if (role[0] == "Application") {
+                return { labelText: "Application", labelColor: null, name: "application-label" };
+            } else {
+                return { labelText: "Internal", labelColor: null, name: "internal-label" };
+            }
+        }
+    };
+
+    return (
+        <>
+            <Forms
+                onSubmit={ () => {
+                    onSubmit({ groups: initialValues?.tempGroupList });
+                } }
+                submitState={ triggerSubmit }
+            >
+                <TransferComponent
```

Comment:
I think it would be better to have sub components for this component.

ex: `TransferList` will be the parent component. Subcomponents could be `TransferList.Item`.